test/e2e_testify/common_test.go

```diff
@@ -0,0 +1,263 @@
+package e2etestify
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	veleroexec "github.com/vmware-tanzu/velero/pkg/util/exec"
+	"github.com/vmware-tanzu/velero/test/e2e"
+)
+
+func installVelero(ctx context.Context) error {
+	if err := e2e.EnsureClusterExists(ctx); err != nil {
+		return errors.Wrap(err, "check cluster exists")
+	}
+
+	// Pass global variables to option parameters.
+	options, err := e2e.GetProviderVeleroInstallOptions(
+		*pluginProvider,
+		*cloudCredentialsFile,
+		*bslBucket,
+		*bslPrefix,
+		*bslConfig,
+		*vslConfig,
+		providerPlugins(*pluginProvider),
+	)
+	if err != nil {
+		return errors.Wrap(err, "get velero install options")
+	}
+
+	options.UseRestic = false
+	options.Features = "EnableAPIGroupVersions"
+	options.Image = *veleroImage
+
+	if err := e2e.InstallVeleroServer(options); err != nil {
+		return errors.Wrap(err, "install velero server")
+	}
+
+	return nil
+}
+
+func providerPlugins(provider string) []string {
+	// TODO: make plugin images configurable
+	switch provider {
+	case "aws":
+		return []string{"velero/velero-plugin-for-aws:v1.1.0"}
+	case "azure":
+		return []string{"velero/velero-plugin-for-microsoft-azure:v1.1.1"}
+	case "vsphere":
+		return []string{"velero/velero-plugin-for-aws:v1.1.0", "velero/velero-plugin-for-vsphere:v1.0.2"}
+	default:
+		return []string{""}
+	}
+}
+
+func InstallCRD(ctx context.Context, crdFile, ns string) error {
+	fmt.Printf("Install CRD %s.\n", crdFile)
+
+	cmd := exec.CommandContext(ctx, "kubectl", "apply", "-f", crdFile)
```

Comment:
This can be addressed in #3133, but I think it would be better to specify the namespace where the CRDs are being installed using the `ns` value passed in rather than having to know the namespace specified in the YAML.